### PR TITLE
[Glimmer] improve text/mustache formatting

### PR DESCRIPTION
**Pull request by GavinJoyce**
Fixes https://github.com/prettier/prettier/issues/6184

**Input:**
```hbs
<p>Your username is @{{name}}</p>
<p>Hi {{firstName}} {{lastName}}</p>
```

**Output before this PR:**
```hbs
<p>
  Your username is @
  {{name}}
</p>
<p>
  Hi
  {{firstName}}
  {{lastName}}
</p>
```

**Output after this PR:**

```hbs
<p>
  Your username is @{{name}}
</p>
<p>
  Hi {{firstName}} {{lastName}}
</p>
```

- [x] I’ve added tests to confirm my change works.
- [x] ~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)~
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/glimmer/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1501,7 +1538,9 @@ singleQuote: true
     )
     hashPair=(does not need a line break due to being under 80 chars long)
   )
-}}{{/block}}
+}}
+
+{{/block}}
```
Comment (evilebottnawi) [inline tests/glimmer/__snapshots__/jsfmt.spec.js.snap]: Looks ugly

Comment (GavinJoyce) [reply tests/glimmer/__snapshots__/jsfmt.spec.js.snap]: fair enough, I'll address this now

Comment (GavinJoyce) [reply tests/glimmer/__snapshots__/jsfmt.spec.js.snap]: While fixing this, these were also fixed:

**before:**

```hbs
{{#block}}{{#block}}hello{{/block}}{{/block}}
{{#block}}hello{{/block}}
```

**after:**

```hbs
{{#block}}
  {{#block}}
    hello
  {{/block}}
{{/block}}
{{#block}}
  hello
{{/block}}
```

**Comment by GavinJoyce** [comment]
@duailibe perhaps you might review this too? I've rebased a bunch of times due to conflicts with the change log as other PRs have landed.

This is the first of many PRs this week to improve glimmer/handlebars support. I'm making my way though Intercom's templates (we've a pretty large app) and addressing issues as I find them. I'm hoping that we can quickly iterate to make Prettier for .hbs good enough for use in real apps

**Comment by GavinJoyce** [comment]
An alternative approach that I'm happy to take is to make a series of changes in a fork and then merge these changes back at some point in future. That might allow me to move faster in the short term

@evilebottnawi @duailibe LMK if that approach is preferable to landing a series of PRs in this repo

**Comment by GavinJoyce** [comment]
FYI, I'm experimenting in a fork so that I can move quickly. I'm hoping that I'll be able to package up some of these tests and changes back into prettier in future

https://github.com/GavinJoyce/prettier/pull/1

**Comment by duailibe** [comment]
@GavinJoyce Sorry I've been focused with other stuff for a while

I have zero experience with Glimmer, so I have no idea if this output works in practice, so I trust you on this 😅 

I'll focus specifically on the code and suggest if anything could be done differently.

As for the separate PRs, they're good when we're fixing stuff, or introducing a new style, so we can revert separately if necessary; in this case, we're trying to get to a stable release with Handlebars support, you can definitely make a single giant PR when you believe it's good enough for a release.

**Pull request merged**